pkgs/applications/virtualization/hyperd/default.nix

```diff
@@ -0,0 +1,68 @@
+# This file was generated by go2nix.
+{ stdenv, go, git, fetchgit, fetchhg, fetchbzr, fetchsvn, devicemapper, libvirt, pkgconfig, btrfs-progs, go-md2man, sqlite, systemd, automake, autoconf }:
+
+stdenv.mkDerivation rec {
+  name = "hyperd-${version}";
+  version = "20170117-${stdenv.lib.strings.substring 0 7 rev}";
+  rev = "0195fadc86e12683820478f4c07dc7a65f867a7f";
+
+  goPackagePath = "github.com/hyperhq/hyperd";
+
+  buildInputs = [ go git devicemapper libvirt pkgconfig btrfs-progs
+    go-md2man sqlite btrfs-progs systemd automake autoconf
+  ];
+
+  src = fetchgit {
+    inherit rev;
+    leaveDotGit = true;
+    deepClone = true;
+    url = "https://github.com/hyperhq/hyperd.git";
+    sha256 = "0d5fkc3vniyh5nkfrna2bfzaxhni6njq0h03rp2x13vm2ha78dy4";
+  };
+
+  unpackPhase = ''
+    export GOPATH=`pwd`
+    mkdir src
+    mkdir -p src/${goPackagePath}
+    cp -r ${src}/* src/${goPackagePath}
+    cp -r ${src}/.git src/${goPackagePath}
+    chmod -R +w *
+
+    #  "unvendorize" stuff.
+    mv src/${goPackagePath}/vendor/github.com/hyperhq/* src/github.com/hyperhq/
+    rmdir src/${goPackagePath}/vendor/github.com/hyperhq
+    mv src/${goPackagePath}/vendor/github.com/* src/github.com/
+    rmdir src/${goPackagePath}/vendor/github.com
+    mv src/${goPackagePath}/vendor/* src/.
+    cd src/${goPackagePath}
+  '';
+
+  configurePhase = ''
+    ./autogen.sh
+    ./configure
+  '';
+
+  buildPhase = ''
+    echo "Go path = $GOPATH"
+    make
+  '';
+
+  preFixup = ''
+    # remove references to go compiler, gcc and glibc
```

Comment:
I guess we have a remove-references feature for this (should be easy to find). If it is not applicable, please elaborate in the code.